app/controllers/audio_files_controller.rb

```diff
@@ -0,0 +1,22 @@
+class AudioFilesController < ApplicationController
+  def new
+    @audio_file = AudioFile.new
+  end
+
+  def create
+    @audio_file = AudioFile.new(audio_file_params)
+    @audio_file.user_id = current_user.id
+    if @audio_file.save
+      flash[:notice] = "Audio file saved successfully"
+      redirect_to dashboard_path
+    else
+      redirect_to new_audio_file_path, notice: @audio_file.errors.messages
```

Comment:
Any reason to do a `redirect` here instead of rendering the `new` template? Its a common pattern, then you can display validation errors inline rather than having to display them in the notice.